### ISSUE TITLE
language isn't stored

### ISSUE THREAD

**Issue by BaumiCoder** [ls1intum/Artemis#5210]
### Describe the bug

If I switch the language to Deutsch, it isn't stored and I have the make this choice at each visit of Artemis.

### To Reproduce

1. Browse to your Artemis server.
2. Change the language to Deutsch
3. Sign in
4. Logout
5. Reload the page :arrow_right:  Page is in English again, also after I login again

### Expected behavior

Artemis remembers the language I had selected. At least after I Sign in.

### What browsers are you seeing the problem on?

Other (specify in "Additional context")

### Additional context

- browsers:
  - Firefox 101.0
  - Opera 87.0.4390.45
- OS: Linux Mint 20.3 64-bit
- Artemis 5.8.3

**Comment by BaumiCoder** [comment]
I think it should update the language key of the user, if the user is signed in. If there is also a language key in the local storage, that maybe triggers the question which of the language keys (local or user) should take action.

Another possible solution in my opinion would be to simplify the language task. Always take the browser language. If Artemis doesn't support the browser language it could fall back to english or an default language of the Artemis server.

**Comment by stefanwaldhauser** [comment]
I would suggest the following approach:
- Currently the language set in the session takes precedence over the one stored in the database. I think we should keep it that way. But: We need to allow a user to set his/her language permanently. 
- Solution: When the user changes the language via the dropdown, ask if this change is only for this current session or if the user details should be updated. Depending on the answer we update only the session storage or also update the language key in the database via a new API endpoint.

**Comment by stefanwaldhauser** [comment]
> I think it should update the language key of the user, if the user is signed in. If there is also a language key in the local storage, that maybe triggers the question which of the language keys (local or user) should take action.
> 
> Another possible solution in my opinion would be to simplify the language task. Always take the browser language. If Artemis doesn't support the browser language it could fall back to english or an default language of the Artemis server.

Good Idea. Maybe something like:
- Source of truth in frontend is language_key in session
**User Not Logged in:**
Opening Artemis:
- If langKey in session -> take it
- If not determine best matching langKey via browser settings and set in session.
Changing Language:
- Change langKey in session to the picked value

**Logging into Artemis:**
- Replace langKey in session with the one from the database
Changing Language:
- Change langKey in session to the picked value and update the database




- User logged in ->

**Comment by BaumiCoder** [comment]
A good and detailed described solution in my opinion. 

I have only one little addition (I don't know if that is technically easy to do or not.):

_user logging into Artemis for the first time:_

- The langKey in the database hasn't been set yet.
- Take the langKey from the session and set it into the database.
- Changing Language:  same as in "Logging into Artemis"

**Comment by stefanwaldhauser** [comment]
@BaumiCoder 
Good Idea! I implemented the feature as a draft in https://github.com/ls1intum/Artemis/pull/5233.
